Library/Homebrew/extend/os/mac/hardware/cpu.rb

```diff
@@ -152,7 +152,11 @@ def sysctl_bool(key)
       end
 
       def sysctl_int(key)
-        sysctl_n(key).to_i
+        if (x = sysctl_n(key).to_i) >= 0
+          x
+        else
+          x & 0xffffffff
+        end
```

Comment:
Could this not just be:

```suggestion
        sysctl_n(key).to_i & 0xffffffff
```

?

Reply:
You're right, thanks!